pydantic/_internal/_validators.py

```diff
@@ -316,6 +322,15 @@ def min_length_validator(x: Any, min_length: Any) -> Any:
             )
         return x
     except TypeError:
+        if isinstance(x, pathlib.PurePath):
+            try:
+                if not (len(str(x)) >= min_length):
+                    raise PydanticKnownError(
+                        'too_short', {'field_type': 'Value', 'min_length': min_length, 'actual_length': len(str(x))}
+                    )
+                return x
+            except TypeError:
+                ...
```

Comment:
As mentioned above, I'd really prefer if this wasn't the long term solution here.